src/net.cpp

```diff
@@ -900,7 +904,21 @@ void CConnman::AcceptConnection(const ListenSocket& hListenSocket) {
         }
     }
 
-    bool whitelisted = hListenSocket.whitelisted || IsWhitelistedRange(addr);
+    CNetPermissionFlags permissionFlags = hListenSocket.permissions;
+    permissionFlags = static_cast<CNetPermissionFlags>(permissionFlags | GetWhitelistPermissionFlags(addr));
```

Comment:
I think adding:

    inline bool HasNetPermissionFlag(CNetPermissionFlags flags, CNetPermissionFlags f)
    {
        return (flags & f) == f;
    }
    inline void AddNetPermissionFlag(CNetPermissionFlags& flags, CNetPermissionFlags f)
    {
        flags = static_cast<CNetPermissionFlags>(flags | f);
    }
    inline void ClearNetPermissionFlag(CNetPermissionFlags& flags, CNetPermissionFlags f)
    {
        flags = static_cast<CNetPermissionFlags>(flags | ~f);
    }

makes using the NetPermFlags a bit clearer fwiw.

Reply:
Not against it, but wouldn't it better to have a templated version somewhere else?

Where could I put such utility function?

Reply:
Or template the bitwise ops directly, maybe -- http://blog.bitwigglers.org/using-enum-classes-as-type-safe-bitmasks/ ? Doesn't seem like there's an existing header file that would fit into to me.

Reply:
It seems to me such change should be better done globally across the project separately, this is not trivial to review.

Reply:
I don't think there's much call for it elsewhere, there's just a couple of `static_cast` for SIGHASH in core_write.cpp and one for ServiceFlags in net.cpp, and some in leveldb. Going by:

    $ grep 'static_cast<[^>*&]*>[(][^)]*[|&~^]' -rI | grep -v qt.*moc | grep -v paymentrequest.pb.cc | grep -v CNetPermissionFlags

Reply:
fair enough, doing it for net_permissions then